plot-and-scatter/dchp3
# New Editing Functionality Architecture

# Summary
# Closes #31 Adds the architecture a proposed Entry Editing solution

## Notes
I think using a popup is our best shot at doing Entry Editing. I considered a lot of alternatives before stumbling onto this one, such as:
1. second edit form
2. passing down an "editable" prop and changing the form based on that (we'll still probably have "editable" as context info, so that we can hide the popup when a user isn't verified)
3. Replacing current Entry with a form that is only mutable if you're in Edit mode

I don't remember what else. But there are a lot of subtle problems to navigate here. The `Popover` element helps us bypass practically all of those additional problems

## Testing
Basically: 
1. Go to any word. 
4. Then, press the red book icon next to the Headword (We'll need to change the icon later).
5. Enter the new Headword you want to replace the current headword with
6. Verify that the headword did indeed change for that entry in the database (not sure if you know about prisma studio or not, but `npx prisma studio` is really really useful)
